### PR TITLE
8325885: GenShen: harmonize with JDK-8325671

### PR DESCRIPTION
**Pull request by ysramakrishna**
/issue JDK-8325885

Harmonize with the changes in JDK-8325671.

Renames:
- GLOBAL_NON_GEN -> NON_GEN
- GLOBAL_GEN -> GLOBAL
- ? -> UNKNOWN

Some reordering of arguments to follow existing convention.

**Testing:**
- [x] GHA (TestHumongous... failures are independent, and addressed in JDK-8327000)
- [x] code pipeline testing

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Issue
 * [JDK-8325885](https://bugs.openjdk.org/browse/JDK-8325885): GenShen: harmonize with JDK-8325671 (**Sub-task** - P4)

### Reviewers
 * [Kelvin Nilsen](https://openjdk.org/census#kdnilsen) (@kdnilsen - Committer) ⚠️ Review applies to [c1d38870](https://git.openjdk.org/shenandoah/pull/402/files/c1d388705d158a27d5a3da4f4d23dd268118d62b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/shenandoah.git pull/402/head:pull/402` \
`$ git checkout pull/402`

Update a local copy of the PR: \
`$ git checkout pull/402` \
`$ git pull https://git.openjdk.org/shenandoah.git pull/402/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 402`

View PR using the GUI difftool: \
`$ git pr show -t 402`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/shenandoah/pull/402.diff">https://git.openjdk.org/shenandoah/pull/402.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/shenandoah/pull/402#issuecomment-1974203271)

### CODE REVIEW

File: src/hotspot/share/gc/shenandoah/shenandoahConcurrentMark.cpp [hunk 1]
```diff
@@ -243,6 +241,8 @@ void ShenandoahConcurrentMark::concurrent_mark() {
   ShenandoahSATBMarkQueueSet& qset = ShenandoahBarrierSet::satb_mark_queue_set();
   ShenandoahFlushSATBHandshakeClosure flush_satb(qset);
   for (uint flushes = 0; flushes < ShenandoahMaxSATBBufferFlushes; flushes++) {
+    // TODO: ysr : hoist this switch out of the loop & refactor ?? How may
```
Comment (kdnilsen) [inline src/hotspot/share/gc/shenandoah/shenandoahConcurrentMark.cpp:244]: The default value of ShenandoahMaxSATBBufferFlushes is 5.  I think the number of iterations is typically low.  Given low number of typical iterations, may not be worth it to restructure the loop control.

**Review by kdnilsen — APPROVED** [review]
Looks good to me.  Thanks.

File: src/hotspot/share/gc/shenandoah/shenandoahConcurrentMark.cpp (hunk 1 above)
Comment (ysramakrishna) [reply src/hotspot/share/gc/shenandoah/shenandoahConcurrentMark.cpp:244]: Good point, I'll clear this comment which was inadvertently left behind here.

Comment (ysramakrishna) [reply src/hotspot/share/gc/shenandoah/shenandoahConcurrentMark.cpp]: Done.

**Pull request closed without merging**